Improved comments, added millis() macro, made example main() more useful for systick_irq example

ADBeta opened the pull request.

prosper00 reviewed (commented):
what do you think about switching the GPIO stuff to funPinMode() and funDigitalWrite()?

ADBeta commented:
Now uses funPinMode etc, tested to work the same

ADBeta commented on examples/systick_irq/systick_irq.c:
```diff
@@ -2,74 +2,81 @@
 #include <stdio.h>
 
-volatile uint32_t systick_cnt;
+#define SYSTICK_CMP_1_MILLISECOND ((uint32_t)FUNCONF_SYSTEM_CORE_CLOCK / 1000)
```
I hope no one ever uses this chip at a HCLK of 1KHz
Usually 48MHz is default so this should be fine, and anyone going below 1000 will have issues more important that this breaking I would think

TommyMurphyTM1234 replied:
I assumed it was an edge case - if even possible at all - but just wanted to ask just in case. :-)

ADBeta replied:
no problem
I'm not sure if it is possible, but if it is it's probably an awful idea anyway haha.

ADBeta commented:
Typos are fixed

TommyMurphyTM1234 commented:
> Typos are fixed

I think I'me looking at the latest changes but still see typos... :-|

ADBeta commented:
Double checked, fixed indents and one missing fix -
Previous fixes were visible on the previous commit, not sure why you couldn't see them :thinking:

TommyMurphyTM1234 commented:
> Double checked, fixed indents and one missing fix - Previous fixes were visible on the previous commit, not sure why you couldn't see them 🤔

I don't understand - when I go into the PR and review the latest changes I still see this for example:

![2024-09-25 17 34 26](https://github.com/user-attachments/assets/37c795ea-0a83-4679-bfbd-7681b6f6d1c3)

ADBeta commented:
> > Double checked, fixed indents and one missing fix - Previous fixes were visible on the previous commit, not sure why you couldn't see them 🤔
> 
> I don't understand - when I go into the PR and review the latest changes I still see this for example:
> 
> ![2024-09-25 17 34 26](https://private-user-images.githubusercontent.com/14550604/370807925-37c795ea-0a83-4679-bfbd-7681b6f6d1c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyODI3MjAsIm5iZiI6MTcyNzI4MjQyMCwicGF0aCI6Ii8xNDU1MDYwNC8zNzA4MDc5MjUtMzdjNzk1ZWEtMGE4My00Njc5LWJmYmQtNzY4MWI2ZjZkMWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI1VDE2NDAyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4NTk5NjY3NGQ0ODAwYjdhZmFkNTVlNWI0MzE1YTkxODBkY2JmZDhjMjY1MWJiYzA1YjU3Y2I3ODBlZWZiMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tihsvZT4GWw68KH9MZ_Y7wz-EYNTW8lYU4nC6TrJtAs)

I missed that one somehow....
I've fixed that one too now, should all be done :crossed_fingers: 
Let me know if you know of a good spellchecker plugin for nvim/vim lol

cnlohr commented:
This all looks good to me now!
